### PR TITLE
Tests | Run tests with config.json and transition to public CI pipelines

### PR DESCRIPTION
**Pull request by cheenamalhotra**
This PR:
- Updates tests to run with `config.json` properties instead of Environment variables.
- Updates scripts to build tests with provided `TargetNetFxVersion` and `TargetNetCoreVersion` msbuild variables.
- Introduces public CI pipelines in public DevOps org.

> **Please Note**: @Wraith2 @ErikEJ @roji (and many other contributors)
This PR changes the way tests are run locally and will require re-configuration. Thanks!

### CODE REVIEW

File: BUILDGUIDE.md [hunk 1]
```diff
@@ -65,32 +65,32 @@ Once the environment is setup properly, execute the desired set of commands belo
 
 ```bash
 > msbuild /t:BuildTestsNetCore
-# Build tests for the .Net Core driver.
+# Build the tests for the .Net Core driver.
```
Comment (v-kaywon) [inline BUILDGUIDE.md]:
```suggestion
# Build the tests for the .Net Core driver. Default .Net Core version is 2.1.
```

File: BUILDGUIDE.md [hunk 2]
```diff
@@ -65,32 +65,32 @@ Once the environment is setup properly, execute the desired set of commands belo
...
 ```
 
 ```bash
 > msbuild /t:BuildTestsNetFx
-# Build tests for the .Net Framework (NetFx) driver.
+# Build the tests for the .Net Framework (NetFx) driver.
```
Comment (v-kaywon) [inline BUILDGUIDE.md]:
```suggestion
# Build the tests for the .Net Framework (NetFx) driver. Default NetFx version is 4.6.
```

File: BUILDGUIDE.md (hunk 1 above)
Comment (ErikEJ) [reply BUILDGUIDE.md]: And it is ".NET"  :smile:

File: src/Microsoft.Data.SqlClient/tests/FunctionalTests/Microsoft.Data.SqlClient.Tests.csproj [hunk 3]
```diff
@@ -45,11 +45,7 @@
     <Compile Include="SqlConnectionTest.RetrieveStatistics.cs" />
     <Compile Include="FakeDiagnosticListenerObserver.cs" />
     <Compile Include="SqlMetaDataTest.cs" />
-    <Compile Include="ExceptionTest.cs" />
```
[the remark this answers is not in the document]
Comment (cheenamalhotra) [reply src/Microsoft.Data.SqlClient/tests/FunctionalTests/Microsoft.Data.SqlClient.Tests.csproj]: Yes, these files are moved to Manual Tests as they need DataTestUtility to work with. Check [here](https://github.com/dotnet/SqlClient/pull/280/files/f5337edb26c7b92be317e32a86eb2521eb088b88?file-filters%5B%5D=.csproj#diff-75a729676a754cbb846b7db654232f44R72)

File: src/Microsoft.Data.SqlClient/tests/ManualTests/DataCommon/DataTestUtility.cs [hunk 4]
```diff
@@ -9,56 +9,90 @@
...
-        public const string AKVKeyName = "TestSqlClientAzureKeyVaultProvider";
-        public static readonly string AKVBaseUrl = null;
-        public static readonly string AKVUrl = null;
-        public static readonly string ClientId = null;
-        public static readonly string ClientSecret = null;
+        public static readonly string NPConnectionString = null;
+        public static readonly string TCPConnectionString = null;
+        public static readonly string AADAccessToken = null;
+        public static readonly string AADPasswordConnectionString = null;
+        public static readonly string AkvBaseUrl = null;
```
[the remark this answers is not in the document]
Comment (cheenamalhotra) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/DataCommon/DataTestUtility.cs]: Renamed back to AKV*

File: src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/ExceptionTest/ConnectionExceptionTest.cs [hunk 5]
```diff
@@ -5,9 +5,9 @@
 using System;
 using Xunit;
 
-namespace Microsoft.Data.SqlClient.Tests
```
[the remark this answers is not in the document]
Comment (cheenamalhotra) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/ExceptionTest/ConnectionExceptionTest.cs]: Dependency on DataTestUtility from Manual tests was used for these 2 classes in Functional Tests. This change is to make both test projects independent of each other and Functional Tests be able to run without any configurations needed.

File: src/Microsoft.Data.SqlClient/tests/FunctionalTests/Microsoft.Data.SqlClient.Tests.csproj [hunk 6]
```diff
@@ -45,11 +45,7 @@
...
-    <Compile Include="TcpDefaultForAzureTest.cs" />
     <Compile Include="SqlConnectionStringBuilderTest.cs" />
-    <!-- <Compile Include="DiagnosticTest.cs" /> -->
```
Comment (David-Engel) [inline src/Microsoft.Data.SqlClient/tests/FunctionalTests/Microsoft.Data.SqlClient.Tests.csproj]: Please keep this line. I'd like to re-enable this test some day when I can get RemoteExecutor working..

File: src/Microsoft.Data.SqlClient/tests/FunctionalTests/Microsoft.Data.SqlClient.Tests.csproj [hunk 7]
```diff
@@ -63,6 +59,7 @@
     <PackageReference Include="microsoft.dotnet.buildtools.testsuite" Version="2.2.0-preview1-03110-01" />
     <PackageReference Include="System.Linq.Expressions" Version="4.3.0" />
     <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="4.5.1" />
+    <PackageReference Include="Newtonsoft.Json" Version="12.0.2" />
   </ItemGroup>
   <ItemGroup Condition=" '$(TargetFramework)' == 'netcoreapp2.1'">
     <PackageReference Include="System.Data.Odbc" Version="4.5.0" />
```
Comment (v-kaywon) [inline src/Microsoft.Data.SqlClient/tests/FunctionalTests/Microsoft.Data.SqlClient.Tests.csproj]: I was able to build .net core 3.0 after adding this condition:
```
<ItemGroup Condition=" '$(TargetFramework)' == 'netcoreapp3.0'">
    <PackageReference Include="System.Data.Odbc" Version="4.6.0" />
  </ItemGroup>
```

Comment (cheenamalhotra) [reply src/Microsoft.Data.SqlClient/tests/FunctionalTests/Microsoft.Data.SqlClient.Tests.csproj]: Seems like building and running tests with netcoreapp3.0 also needs some fixes. Leaving this change as it needs to be addressed specifically.

**Pull request merged**